File: adventofcodeday1.py
# import sys

# count = 0
# dial = 50


# def fix_current_rotation(num):
    


# def rotate(direction):
#     global dial
#     if direction.startswith("L"):
#         dial -= int(direction[1:])
#         check(fix_current_rotation(dial))
#     else:
#         dial += int(direction[1:])
#         check(fix_current_rotation(dial))


# def check(num):
#     global count
#     if num == 0:
#         count += 1


# def main(input_text):
#     commands = input_text.strip().split('\n')
#     for command in commands:
#         rotate(command.strip())
#     print(count)


# if __name__ == "__main__":
#     with open('input.txt', 'r') as f:
#         input_lines = f.read()
#     main(input_lines)

File: problems/adventofcodeday1.js
let count = 0
let dial = 50
function fix_current_rotation(num) { 
    const modulo = num % 100
    if (modulo < 0) {
        return (100 + modulo);
    } else {
        return modulo;
    }
}
function rotate(direction) {
    if (direction.startsWith("L")) {
        check(fix_current_rotation(dial -= Number(direction.slice(1))))
    } else { 
        check(fix_current_rotation(dial += Number(direction.slice(1))))
    }
}
function check(num){ 
    if (num == 0){
        count++
    }
}
function main(input) { 
    const commands = input.trim().split('\n');
    commands.forEach(command => rotate(command.trim()));
    console.log(count);
}
const fs = require('fs');
const input = fs.readFileSync('input.txt', 'utf8');
main(input);